app/code/community/Bolt/Boltpay/Model/Observer.php

```diff
@@ -74,38 +74,64 @@ public function clearCartCacheOnOrderCanceled($observer) {
      */
     public function setSuccessSessionData($observer) {
 
+        /** @var Mage_Checkout_Model_Session $checkoutSession */
+        $checkoutSession = Mage::getSingleton('checkout/session');
         $requestParams = Mage::app()->getRequest()->getParams();
 
         // Handle only Bolt orders
-        if (!isset($requestParams['bolt_payload'])) {
+        if (!isset($requestParams['bolt_payload'])  && !isset($requestParams['bolt_transaction_reference'])) {
```

Comment:
@matt-thomason 
It should be `&&`.  Neither of the conditions should be true for an early return.  And in practice, only one of the two things will be true for Bolt orders.

If I wanted to wrap the whole method body instead of doing an early return, I could use `||` like this
```
if (isset($requestParams['bolt_payload']) || isset($requestParams['bolt_transaction_reference'])){
     // ... all the rest of the method
}
```

But, there has been a lot of preference in our code for the early return pattern, so it is done the way above for this reason.

Reply:
I misread the lack of `!` in the following if statements, got it. Reviewed too late in the night, sorry 👍

Reply:
i feel that late night working.